src/command_modules/azure-cli-network/azure/cli/command_modules/network/commands.py

```diff
@@ -349,6 +351,28 @@ def _make_singular(value):
                                      custom_func_name='update_ag_{}'.format(_make_singular(subresource)),
                                      child_collection_prop_name=subresource, validator=create_validator)
 
+    with self.command_group('network application-gateway rewrite-rule', network_ag_sdk, min_api='2018-12-01') as g:
+        g.custom_command('create', 'create_ag_rewrite_rule', supports_no_wait=True)
+        g.custom_command('show', 'show_ag_rewrite_rule')
+        g.custom_command('list', 'list_ag_rewrite_rules')
+        g.custom_command('delete', 'delete_ag_rewrite_rule', supports_no_wait=True)
+        g.generic_update_command('update', command_type=network_ag_sdk, supports_no_wait=True,
+                                 custom_func_name='update_ag_rewrite_rule',
+                                 child_collection_prop_name='rewrite_rule_sets.rewrite_rules',
+                                 child_collection_key='name.name',
```

Comment:
1 & 2 - yes. This PR adds the ability to target nested collections. For 3, those should really be public methods...

Reply:
I think it could be beneficial to mention the helper functions. They seem like they would be very useful to the advanced user working with nested collections.

Reply:
They are. I will make the change in this PR.